modules/blocks/chain.js

```diff
@@ -340,26 +337,21 @@ Chain.prototype.applyBlock = function(block, saveBlock, cb) {
 				new Promise((resolve, reject) => {
 					modules.accounts.setAccountAndGet(
 						{ publicKey: transaction.senderPublicKey },
-						(err, sender) => {
+						(accountErr, sender) => {
 							// DATABASE: write
 							modules.transactions.applyUnconfirmed(
```

Comment:
So no reason to perform `modules.transactions.applyUnconfirmed` when `accountErr` happens. 
Extract error handling to a separate function:
```
	err = `Failed to apply transaction: ${
	 transaction.id
	} '-' ${err || accountErr}`;
	library.logger.error(err);
	library.logger.error('Transaction', transaction);
```
and call it in 2 places.

Reply:
Makes sense.